src/test/regress/expected/multi_router_planner.out

```diff
@@ -0,0 +1,1072 @@
+	FROM articles_hash
+	WHERE author_id = 1;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+ id | word_count |          avg          
+----+------------+-----------------------
+ 11 |       1347 | 1347.0000000000000000
+ 21 |       5890 | 3618.5000000000000000
+ 31 |       7271 | 4836.0000000000000000
+  1 |       9572 | 6020.0000000000000000
+ 41 |      11814 | 7178.8000000000000000
+(5 rows)
+
+-- window functions are not supported for not router plannable queries
+SELECT id, MIN(id) over (order by word_count)
+	FROM articles_hash
+	WHERE author_id = 1 or author_id = 2;
+ERROR:  cannot perform distributed planning on this query
+DETAIL:  Window functions are currently unsupported
+	
+-- but they are not supported for not router plannable queries
+SELECT LAG(title, 1) over (ORDER BY word_count) prev, title, word_count 
+	FROM articles_hash
+	WHERE author_id = 5 or author_id = 1;
+ERROR:  cannot perform distributed planning on this query
+DETAIL:  Window functions are currently unsupported
+-- complex query hitting a single shard 	
+SELECT
+	count(DISTINCT CASE
+			WHEN 
+				word_count > 100
+			THEN
+				id
+			ELSE
+				NULL
+			END) as c
+	FROM
+		articles_hash
+	WHERE
+		author_id = 5;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+ c 
+---
+ 5
+(1 row)
+
+-- same query is not router plannable if hits multiple shards
+SELECT
+	count(DISTINCT CASE
+			WHEN 
+				word_count > 100
+			THEN
+				id
+			ELSE
+				NULL
+			END) as c
+	FROM
+		articles_hash
+ 	GROUP BY
+		author_id;
+ERROR:  cannot compute aggregate (distinct)
+DETAIL:  aggregate (distinct) on complex expressions is unsupported
+HINT:  You can load the hll extension from contrib packages and enable distinct approximations.
+-- queries inside transactions can be router plannable
+BEGIN;
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1
+	ORDER BY id;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+ id | author_id |    title     | word_count 
+----+-----------+--------------+------------
+  1 |         1 | arsenous     |       9572
+ 11 |         1 | alamo        |       1347
+ 21 |         1 | arcading     |       5890
+ 31 |         1 | athwartships |       7271
+ 41 |         1 | aznavour     |      11814
+(5 rows)
+
+END;
+-- cursor queries inside transactions are not router plannable
+BEGIN;
+DECLARE test_cursor CURSOR FOR 
+	SELECT *
+		FROM articles_hash
+		WHERE author_id = 1
+		ORDER BY id;
+DEBUG:  predicate pruning for shardId 103301
+FETCH test_cursor;
+ id | author_id |  title   | word_count 
+----+-----------+----------+------------
+  1 |         1 | arsenous |       9572
+(1 row)
+
+FETCH test_cursor;
+ id | author_id | title | word_count 
+----+-----------+-------+------------
+ 11 |         1 | alamo |       1347
+(1 row)
```

Comment:
> > +	SELECT *
> > +		FROM articles_hash
> > +		WHERE author_id = 1
> > +		ORDER BY id;
> > +DEBUG:  predicate pruning for shardId 103301
> > +FETCH test_cursor;
> > + id | author_id |  title   | word_count 
> > +----+-----------+----------+------------
> > +  1 |         1 | arsenous |       9572
> > +(1 row)
> > +
> > +FETCH test_cursor;
> > + id | author_id | title | word_count 
> > +----+-----------+-------+------------
> > + 11 |         1 | alamo |       1347
> > +(1 row)
> 
> Very minor: do you think should we close the cursor

Cursors are automatically closed at the end of a transaction (or
session in the case of WITH HOLD).